iceoryx_utils/include/iceoryx_utils/internal/cxx/convert.inl

```diff
@@ -20,6 +20,21 @@ namespace iox
 {
 namespace cxx
 {
+template <>
+inline typename std::enable_if<!std::is_convertible<uint8_t, std::string>::value, std::string>::type
+convert::toString(const uint8_t& t)
+{
+    return toString(static_cast<uint16_t>(t));
+}
```

Comment:
Nice. Could you please add a comment at the top why this is necessary. In half a year we might have forgotten about the reason and someone might do a cleanup and remove it.